### PR TITLE
Feature/chaos api

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/org/radix/universe/system/RadixSystem.java [hunk 1]
```diff
@@ -137,14 +138,14 @@ void setJsonAgent(Map<String, Object> props) {
 	// FIXME: Should serialize ECKeyPair directly
 	@JsonProperty("key")
 	@DsonOutput(Output.ALL)
-	byte[] getJsonKey() {
-		return (key == null) ? null : key.getBytes();
+	String getJsonKey() {
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/org/radix/universe/system/RadixSystem.java:141]: Not sure why this was changed?

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/jsonrpc/RadixJsonRpcServer.java [hunk 2]
```diff
@@ -256,6 +266,22 @@ private JSONObject handle(JSONObject jsonRpcRequest) {
 						result = new JSONObject().put("status", atomStatus.toString());
 					}
 					break;
+				case "Chaos.enableMessageFlooder":
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/jsonrpc/RadixJsonRpcServer.java:269]: As mentioned previously, I don't believe it is a good idea to include the sorts of things that have the potential to break the node in the JSON RPC processing pipeline, as it requires stateful stream inspection to firewall them off with nginx.

@rvelaz probably best if you weigh in here from an operational perspective.

**Review by clstrfsck — APPROVED** [review]
As commented, not sure that we should be providing node-breaking APIs through JSON RPC.

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java [hunk 3]
```diff
@@ -19,23 +19,28 @@
...
 import com.google.common.io.CharStreams;
 import com.radixdlt.consensus.QuorumCertificate;
-import com.radixdlt.middleware2.store.CommandToBinaryConverter;
 import com.radixdlt.network.addressbook.AddressBook;
 import com.radixdlt.properties.RuntimeProperties;
 import com.radixdlt.serialization.DsonOutput;
 import com.radixdlt.serialization.Serialization;
 import com.radixdlt.store.LedgerEntryStore;
 import com.radixdlt.universe.Universe;
+import com.radixdlt.utils.Base58;
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java:43]: Doesn't seem to be used?

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java [hunk 4]
```diff
@@ -19,23 +19,28 @@
 
 import com.google.inject.Inject;
 import com.radixdlt.ModuleRunner;
+import com.radixdlt.chaos.MessageFloodUpdate;
+import com.radixdlt.consensus.bft.BFTNode;
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java:23]: Doesn't seem to be used?

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java [hunk 5]
```diff
@@ -19,23 +19,28 @@
...
+import com.radixdlt.crypto.ECPublicKey;
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java:24]: Doesn't seem to be used?

File: radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java [hunk 6]
```diff
@@ -19,23 +19,28 @@
...
 import com.radixdlt.crypto.Hasher;
 import com.radixdlt.consensus.bft.VerifiedVertex;
+import com.radixdlt.crypto.exception.PublicKeyException;
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java:27]: Doesn't seem to be used?

File: radixdlt-core/radixdlt/src/test/java/com/radixdlt/chaos/MessageFloodUpdateTest.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package com.radixdlt.chaos;
```
Comment (clstrfsck) [inline radixdlt-core/radixdlt/src/test/java/com/radixdlt/chaos/MessageFloodUpdateTest.java:1]: Missing header comment.

File: radixdlt-core/radixdlt/src/main/java/org/radix/GlobalInjector.java [hunk 8]
```diff
@@ -180,6 +181,8 @@ LocalSystem localSystem(
 			new AddressBookModule(),
 			new HostIpModule(properties),
 
+			new ChaosModule(),
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/GlobalInjector.java:184]: Are we going to have this module permanently present?

File: radixdlt-core/radixdlt/src/main/java/org/radix/Radix.java [hunk 9]
```diff
@@ -153,6 +153,9 @@ public static void start(RuntimeProperties properties) {
 		final ModuleRunner mempoolReceiverRunner = moduleRunners.get("mempool");
 		mempoolReceiverRunner.start();
 
+		final ModuleRunner chaosRunner = moduleRunners.get("chaos");
+		chaosRunner.start();
```
Comment (siy) [inline radixdlt-core/radixdlt/src/main/java/org/radix/Radix.java:157]: Is it necessary to have ChaosRunner working in the regular node?

**Pull request merged**